sdk/identity/identity/src/util/logging.ts

```diff
@@ -1,9 +1,131 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { createClientLogger } from "@azure/logger";
+import { createClientLogger, AzureLogger } from "@azure/logger";
+import { TokenCredentialOptions } from "../client/identityClient";
+import { CredentialUnavailable } from "../client/errors";
 
 /**
  * The AzureLogger used for all clients within the identity package
  */
 export const logger = createClientLogger("identity");
+
+interface EnvironmentAccumulator {
+  missing: string[];
+  assigned: string[];
+}
+
+/**
+ * Separates a list of environment variable names into a plain object with two arrays: an array of missing environment variables and another array with assigned environment variables.
+ * @param supportedEnvVars List of environment variable names
+ */
+export function processEnvVars(supportedEnvVars: string[]): EnvironmentAccumulator {
+  return supportedEnvVars.reduce(
+    (acc: EnvironmentAccumulator, envVariable: string) => {
+      if (process.env[envVariable]) {
+        acc.assigned.push(envVariable);
+      } else {
+        acc.missing.push(envVariable);
+      }
+      return acc;
+    },
+    { missing: [], assigned: [] }
+  );
+}
+
+/**
+ * Based on a given list of environment variable names,
+ * logs the environment variables currently assigned during the usage of a credential that goes by the given name.
+ * Logs nothing if the log level is "warning" or "error".
+ * @param credentialName Name of the credential in use
+ * @param supportedEnvVars List of environment variables supported by that credential
+ */
+export function logEnvVars(credentialName: string, supportedEnvVars: string[]): void {
+  // Logs nothing if the log level is "warning" or "error".
+  const logLevel = process.env.AZURE_LOG_LEVEL;
```

Comment:
this isn't the only way to set the log level though -- someone could set it with the function rather than the environment variable and this wouldn't log

Reply:
An easier way to test this is if you have the logger in question you can do `if(logger.warning.enabled || logger.error.enabled)`

Reply:
Ohh! Thank you. I wasn't aware of that approach 🌞

Reply:
I'll remove the conditionals here. Ty ty